### PR TITLE
fix: apply correct UTC offsets for RFC 822 timezone abbreviations

### LINKED ISSUE #237: RSS "pubDate", DST timezone identifiers not handled
DST timezone identifiers are not currently being handled properly in RFC 822 formatted dates.

Take the [CBC RSS Feeds](https://www.cbc.ca/rss/) as an example. `Mon, 21 Apr 2025 06:00:00 EDT` is wrongly parsed as being UTC, which leads to items being marked as 4 hours later they actually are.

[RFC 822 does allow the use of defined offsets as timezone identifiers](https://datatracker.ietf.org/doc/html/rfc822#section-5):

```
 zone        =  "UT"  / "GMT"                ; Universal Time
                                                 ; North American : UT
                 /  "EST" / "EDT"                ;  Eastern:  - 5/ - 4
                 /  "CST" / "CDT"                ;  Central:  - 6/ - 5
                 /  "MST" / "MDT"                ;  Mountain: - 7/ - 6
                 /  "PST" / "PDT"                ;  Pacific:  - 8/ - 7
                 /  1ALPHA                       ; Military: Z = UT;
                                                 ;  A:-1; (J not used)
                                                 ;  M:-12; N:+1; Y:+12
                 / ( ("+" / "-") 4DIGIT )        ; Local differential
                                                 ;  hours+min. (HHMM)
```

Now, one could argue that using RFC 822 in 2025 is a bit stupid, and I'd agree, but I don't work at CBC unfortunately. The issue has been reported to them and their reply was that their RSS feeds are not a priority.

Would you be open to adding a fix to `dateparser.go` to handle this edge-case? I can contribute a PR.

### PR DESCRIPTION
**Pull request by lawrence3699**
Fixes #237

**Problem:** RSS dates with named timezone abbreviations like EDT, CDT, PDT, etc. are parsed with UTC offset 0 instead of the correct offset. For example, `Mon, 21 Apr 2025 06:00:00 EDT` is treated as UTC rather than UTC-4.

This happens because Go's `time.Parse` recognises named zones but assigns offset 0, and `time.LoadLocation` rejects short abbreviation names like "EDT".

**Fix:** Add a lookup table mapping the ten RFC 822 §5 timezone abbreviations (UT, GMT, EST, EDT, CST, CDT, MST, MDT, PST, PDT) to their fixed UTC offsets. After parsing, if the zone name matches the table and the offset is 0, re-parse with `time.ParseInLocation` using the correct fixed-offset location.

**Before:**
```
ParseDate("Mon, 21 Apr 2025 06:00:00 EDT") → offset 0 (UTC)
```

**After:**
```
ParseDate("Mon, 21 Apr 2025 06:00:00 EDT") → offset -14400 (UTC-4)
```

**Validation:**
- `go test ./internal/shared/ -v` — all pass
- `go test ./rss/ ./atom/ ./json/ ./extensions/` — all pass
- New test covers all 8 non-UTC abbreviations, verifies local hour preserved and UTC hour correct
- Edge-case tests for numeric offsets and GMT confirm no regressions

### CODE REVIEW

File: internal/shared/dateparser.go [hunk 1]
```diff
@@ -138,6 +138,23 @@ var dateFormats = []string{
...
+	"EST": -5 * 3600,
+	"EDT": -4 * 3600,
+	"CST": -6 * 3600,
+	"CDT": -5 * 3600,
+	"MST": -7 * 3600,
+	"MDT": -6 * 3600,
+	"PST": -8 * 3600,
+	"PDT": -7 * 3600,
+}
+
```
Comment (infogulch) [inline internal/shared/dateparser.go:157]:
```suggestion
// rfc822Zones maps RFC 822 §5 timezone abbreviations to their corresponding time.Location.
// Go's time.Parse recognises the abbreviation but assigns offset 0,
// and time.LoadLocation rejects short names such as "EDT", so we need an
// explicit table.
var rfc822Zones = map[string]*time.Location{
	"UT":  time.FixedZone("UT", 0), // included for completeness; offset-0 entries are no-ops in fixRFC822Zone
	"GMT": time.FixedZone("GMT", 0),
	"EST": time.FixedZone("EST", -5*3600),
	"EDT": time.FixedZone("EDT", -4*3600),
	"CST": time.FixedZone("CST", -6*3600),
	"CDT": time.FixedZone("CDT", -5*3600),
	"MST": time.FixedZone("MST", -7*3600),
	"MDT": time.FixedZone("MDT", -6*3600),
	"PST": time.FixedZone("PST", -8*3600),
	"PDT": time.FixedZone("PDT", -7*3600),
}
```

**Review by infogulch — COMMENTED** [review]
I like this idea, but it could be simplified.

1. Change `rfc822Zones` to a `map[string]*time.Location`, like the comment suggestion below
2. Eliminate the `fixRFC822Zone` function
3. Change `ParseDate` to loop over `dateFormatsWithNamedZone` *first*
4. In the `time.LoadLocation` failure case, attempt to fall back to a matching zoneName in the map

This eliminates the function and only adds 2 functional lines to ParseDate: https://go.dev/play/p/q40afg8RXha

3&4 might look like this:

```go
	for _, f := range dateFormatsWithNamedZone {
		t, err = time.Parse(f, d)
		if err != nil {
			continue
		}
		zoneName := t.Location().String()

		// This is a format match! Now try to load the timezone name
		loc, err := time.LoadLocation(zoneName)
		if err != nil {
			if rfc822zone, ok := rfc822Zones[zoneName]; ok {
				loc = rfc822zone
			} else {
				// We couldn't load the TZ name. Just use UTC instead...
				return t, nil
			}
		}

		if t, err = time.ParseInLocation(f, ds, loc); err == nil {
			return t, nil
		}
		// This should not be reachable
	}
	for _, f := range dateFormats {
		if t, err = time.Parse(f, d); err == nil {
			return
		}
	}
```

**Comment by infogulch** [comment]
@lawrence3699 any particular reason why you closed your PR?
